### DISCUSSION TITLE
execution times of adjacency_list are different on python and c++

### DISCUSSION THREAD

**Discussion by tianhaoxie** [libigl/libigl#2177] [Q&A] [answered]
Hi, I tested the execution time of adjacency_list on python and c++, but they showed a large difference. The code is shown below:
for python:
`v,_,_,f,_,_=igl.read_obj(path)`
`a = time.time()`
`adj = igl.adjacency_list(f)`
`b = time.time()`
`print((b-a)*1000)`
for c++:
`auto t_start = std::chrono::high_resolution_clock::now();`
 ` igl::readOBJ("../extracted_mesh_puppet.obj", V, F);`
 ` igl::adjacency_list(F,A);`
  `auto t_end = std::chrono::high_resolution_clock::now();`
  `std::chrono::duration<double, std::milli> duration = t_end-t_start;`
 ` std::cout<<duration.count() <<'\n';`

For python, I got the output of around 8 ms, however, in c++ I got an output around 40ms. I think the result suppose to be similar. If you know what cause that or what's the problem of my measurement, please discuss here. Thanks for your time.

**Comment by alecjacobson** [comment, marked as answer]
Is the C++ code compiled in release mode? (with `-O3` and `-DNDEBUG` ?)

**Reply by alecjacobson** [reply]
oh, I think in C++ you're also timing `readOBJ`...

**Reply by tianhaoxie** [reply]
Thanks, after I compile as release mode it gets right.
